lib/AST/ASTImporter.cpp

```diff
@@ -6932,9 +6935,9 @@ Decl *ASTImporter::Import(Decl *FromD) {
   assert((Pos == ImportedDecls.end() || Pos->second == ToD) && "Try to import an already imported Decl");
 
   // Record the imported declaration.
-  ImportedDecls[FromD] = ToD;
+  if (Pos == ImportedDecls.end())
```

Comment:
Is adding this check related to friend importing?

Reply:
To some extent yes: The copy of the IdentifierNamespace is done at new place (in `Imported`) and here it is more safe to call `Imported` too (but is needed only if a call to `Imported` was forgotten after an object was created). The same check is done in `Imported` too. Maybe it is enough here to only call `Imported` without this check and omit the assert too.